data_handler/handlers/health_ratio_level/health_ratio_handlers.py

```diff
@@ -0,0 +1,304 @@
+from handlers.liquidable_debt.utils import Prices
+from handlers.liquidable_debt.values import (USER_FIELD_NAME, HEALTH_FACTOR_FIELD_NAME,
+                                             TIMESTAMP_FIELD_NAME, PROTOCOL_FIELD_NAME)
+
+
+class BaseHealthRatioHandler:
+    """
+    A base handler class that collects data from DB,
+    computes health_ratio level and stores it in the database.
+
+    :cvar CONNECTOR: A DB connection object.
+    """
+    CONNECTOR = DBConnector()
+
+    def __init__(self, state_class: State = None, loan_entity_class: LoanEntity = None):
+        self.state_class = state_class
+        self.loan_entity_class = loan_entity_class
+
+    def fetch_data(self, protocol_name: str) -> tuple:
+        """
+        Prepares the data for the given protocol.
+        :param protocol_name: Protocol name.
+        :return: tuple
+        """
+        fetched_data = self.get_data_from_db()
+        interest_rate_models = self.get_interest_rate_models_from_db(
+            protocol_id=protocol_name
+        )
+
+        return fetched_data, interest_rate_models
+
+    @classmethod
+    def get_interest_rate_models_from_db(cls, protocol_id: str) -> dict:
+        """
+        Returns interest rate models data from DB.
+        :param protocol_id: str
+        :return: dict
+        """
+        return cls.CONNECTOR.get_last_interest_rate_record_by_protocol_id(protocol_id=protocol_id)
+
+    @classmethod
+    def get_data_from_db(cls) -> dict:
+        """
+        Gets the data from the database based on the protocol name.
+        :return: The data from the database.
+        """
+        return cls.CONNECTOR.get_latest_block_loans()
+
+    @classmethod
+    def write_to_db(cls, data: HealthRatioLevel = None) -> None:
+        """
+        Writes the data into the database.
+        :param data: A dictionary of the parsed data.
+        :return: None
+        """
+        cls.CONNECTOR.write_to_db(data)
+
+    @staticmethod
+    def health_ratio_is_valid(health_ratio_level: Decimal = None) -> bool:
+        """
+        Checks if the given health ratio level is valid.
+        :param health_ratio_level: Health ratio level
+        :return: bool
+        """
+        return (health_ratio_level > Decimal("0") and
+                health_ratio_level != Decimal("Infinity"))
+
+
+class ZkLendHealthRatioHandler(BaseHealthRatioHandler):
+    """
+    A zkLend handler that collects data from DB,
+    computes health_ratio level and stores it in the database.
+
+    :cvar CONNECTOR: A DB connection object.
+    """
+
+    def __init__(self):
+        super().__init__(state_class=ZkLendState, loan_entity_class=ZkLendLoanEntity)
+
+    def calculate_health_ratio(self) -> defaultdict:
+        """
+        Calculates health ratio based on provided data.
+        :return: A dictionary of the ready health ratio data.
+        """
+        data, interest_rate_models = self.fetch_data(protocol_name=ProtocolIDs.ZKLEND.value)
```

Comment:
Split this method into 2 methods:
1. method to initialize `loan_entity`
2. Second calculate and get result